pinot-broker/src/main/java/org/apache/pinot/broker/routing/BrokerRoutingManager.java

```diff
@@ -418,17 +420,73 @@ public synchronized void includeServerToRouting(String instanceId) {
   }
 
   /**
-   * Builds/rebuilds the routing for the physical table, for logical tables it is skipped.
+   * Builds/rebuilds the routing for the physical or logical table
    * @param physicalOrLogicalTable a physical table with type or logical table name
    */
   public synchronized void buildRouting(String physicalOrLogicalTable) {
-    // skip route building for logical tables
     if (ZKMetadataProvider.isLogicalTableExists(_propertyStore, physicalOrLogicalTable)) {
-      LOGGER.info("Skipping route building for logical table: {}", physicalOrLogicalTable);
+      buildRoutingForLogicalTable(physicalOrLogicalTable);
+    } else {
```

Comment:
Is the if/else condition still required ? Can the code be moved back to the previous state where there is a short-circuit for logical tables (due to an update in broker resource) ?

Reply:
No. The helix message handler is for update only. At table create, brokerResource is updated so the buildRouting would be called directly.